packages/modeldb-durable-objects/src/encoding.ts

```diff
@@ -0,0 +1,251 @@
+import * as json from "@ipld/dag-json"
+
+import type {
+	Model,
+	ModelValue,
+	PrimaryKeyProperty,
+	PrimaryKeyValue,
+	PrimitiveProperty,
+	PrimitiveValue,
+	PropertyValue,
+	ReferenceProperty,
+} from "@canvas-js/modeldb"
+
+import { assert, mapValues, signalInvalidType } from "@canvas-js/utils"
```

Comment:
Note that DOs use ArrayBuffers to return bytes.

Also, many of the helpers here could be imported from modeldb-sqlite.